File: src/main/java/net/thenamedev/legendapi/gamemanager/GamePlayer.java
package net.thenamedev.legendapi.gamemanager;

import net.thenamedev.legendapi.core.emeralds.EmeraldPlayer;
import net.thenamedev.legendapi.core.exceptions.PlayerNotOnlineException;
import org.bukkit.entity.Player;

/**
 * Created on 5/23/2015
 *
 * @author ThePixelDev
 */
public class GamePlayer {

    private Player p;
    private GamePlayerState state;
    private EmeraldPlayer ep;

    public GamePlayer(Player p, GamePlayerState state) {
        this.p = p;
        this.ep = new EmeraldPlayer(p);
        this.state = state;
    }

    public Player getPlayer() {
        if(p == null) throw new PlayerNotOnlineException();
        return p;
    }

    public GamePlayerState getState() {
        return state;
    }

    public void setState(GamePlayerState state) {
        this.state = state;
    }

    public EmeraldPlayer getEmeralds() {
        return ep;
    }

}

File: src/main/java/legendarena/api/gamemanager/GamePlayer.kt
package legendarena.api.gamemanager

import org.bukkit.entity.Player

class GamePlayer {

    private var p: Player? = null
    private var state: GamePlayerState? = null

    constructor(p: Player, state: GamePlayerState) {
        this.p = p
        this.state = state
    }

    /**
     * Get the player instance
     * @return The player instance
     */
    fun getPlayer(): Player {
        return p!!
    }

    /**
     * Get the state of the player
     * @return The player's state
     * @see GamePlayerState
     */
    fun getState(): GamePlayerState {
        return state!!
    }

    /**
     * Set the state of the player
     * @param state The state to set to
     */
    fun setState(state: GamePlayerState) {
        this.state = state
        if(state == GamePlayerState.SPECTATOR) {
            getPlayer().spigot().collidesWithEntities = false
        }
    }

}